docs/api/python.md

```diff
@@ -60,33 +60,67 @@ print(con.fetchall())
+```python
+import pandas as pd
+my_dictionary = {}
+my_dictionary['test_df'] = pd.DataFrame.from_dict({"i":[1, 2, 3, 4], "j":["one", "two", "three", "four"]})
+con.register('test_df_view', my_dictionary['test_df'])
 con.execute('SELECT * FROM test_df_view')
 con.fetchall()
 # [(1, 'one'), (2, 'two'), (3, 'three'), (4, 'four')]
 ```
 
-You can now use the registered view to create a persistent table in DuckDB:
+Pyarrow tables work in much the same way:
 ```python
-con.execute('CREATE TABLE test_df_table AS SELECT * FROM test_df_view')
+import pyarrow as pa
+arrow_table = pa.Table.from_pydict({'i':[1,2,3,4], 'j':["one", "two", "three", "four"]})
+con.execute('SELECT * FROM arrow_table')
+con.fetchall()
 ```
-> DuckDB keeps a reference to the Pandas data frame after registration. This prevents the data frame from being garbage-collected by the Python runtime. The reference is cleared when the connection is closed, but can also be cleared manually using the `unregister()` method.
+
+You can also create a persistent table in DuckDB from the contents of the DataFrame (or the view):
+```python
+con.execute('CREATE TABLE test_df_table AS SELECT * FROM test_df')
+```
+> DuckDB keeps a reference to the Pandas DataFrame or Arrow object after registration. This prevents them from being garbage-collected by the Python runtime. The reference is cleared when the connection is closed, but can also be cleared manually using the `unregister()` method.
 
 When retrieving the data from DuckDB back into Python, the standard method of calling `fetchall()` is inefficient as individual Python objects need to be created for every value in the result set. When retrieving a lot of data, this can become very slow.
 
-In DuckDB, there are two additional methods that can be used to efficiently retrieve data: `fetchnumpy()` and `fetchdf()`. `fetchnumpy()` fetches the data as a dictionary of `NumPy` arrays. `fetchdf()` fetches the data as a Pandas DataFrame.
+DuckDB's Python client provides multiple additional methods that can be used to efficiently retrieve data.
+### NumPy
+* `fetchnumpy()` fetches the data as a dictionary of NumPy arrays
+
+### Pandas
+* `df()` fetches the data as a Pandas DataFrame
+* `fetchdf()` is an alias of `df()`
+* `fetch_df()` is an alias of `df()`
+* `fetch_df_chunk(vector_multiple)` fetches a portion of the results into a DataFrame. The number of rows returned in each chunk is the vector size (1024 by default) * vector_multiple (1 by default).
 
-Below is an example of using this functionality:
+### Apache Arrow
+* `arrow()` fetches the data as an [Arrow table](https://arrow.apache.org/docs/python/generated/pyarrow.Table.html)
+* `fetch_arrow_table()` is an alias of `arrow()`
+* `fetch_record_batch(chunk_size)` returns an [Arrow record batch reader](https://arrow.apache.org/docs/python/generated/pyarrow.ipc.RecordBatchStreamReader.html) with `chunk_size` rows per batch 
```

Comment:
I just checked this one is actually ```fetch_arrow_reader(chunk_size)```

Otherwise I think it looks pretty good!

Reply:
I'm on version 0.3.3-dev1409 and it looks like fetch_record_batch is working and fetch_arrow_reader is failing with:
```python
AttributeError: 'duckdb.DuckDBPyConnection' object has no attribute 'fetch_arrow_reader'
```